sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -96,6 +98,12 @@ trait CheckAnalysis extends PredicateHelper {
       case InsertIntoStatement(u: UnresolvedRelation, _, _, _, _) =>
         failAnalysis(s"Table not found: ${u.multipartIdentifier.quoted}")
 
+      case a: AlterTableStatement if isTempView(a.tableName) =>
```

Comment:
for v1 commands, only `ALTER TABLE ADD COLUMN` checks temp view and gives a better error message. Here I improve it a little bit to apply it to all ALTER TABLE statements.